Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -862,14 +862,13 @@ class BrowserViewController: UIViewController {
 
     override func observeValueForKeyPath(keyPath: String?, ofObject object: AnyObject?, change: [String: AnyObject]?, context: UnsafeMutablePointer<Void>) {
         let webView = object as! WKWebView
-        if webView !== tabManager.selectedTab?.webView {
-            return
-        }
         guard let path = keyPath else { assertionFailure("Unhandled KVO key: \(keyPath)"); return }
         switch path {
         case KVOEstimatedProgress:
-            guard let progress = change?[NSKeyValueChangeNewKey] as? Float else { break }
-            urlBar.updateProgressBar(progress)
+            if webView == tabManager.selectedTab?.webView {
+                guard let progress = change?[NSKeyValueChangeNewKey] as? Float else { break }
+                urlBar.updateProgressBar(progress)
+            }
         case KVOLoading:
             guard let loading = change?[NSKeyValueChangeNewKey] as? Bool else { break }
             toolbar?.updateReloadStatus(loading)
```

Comment:
We'll want the selected tab check here, too, since `toolbar` and `urlBar` are shared between all tabs. If we don't check the selected tab here, a background tab would cause the stop/reload buttons to change, which wouldn't make sense since those buttons only work for the active tab.